wordpress-mobile/WordPress-Android
# Prepublishing Nudges : create core bottom sheet functionality

Fixes #11602
## Findings

The objective of this PR is to create the core structure that's required to support Prepublishing Nudges within the Android app. 

The behavior in Android, will operate and look similarly to the designs below that are done for iOS. 

<img width="610" alt="Screen Shot 2020-04-09 at 1 20 54 PM" src="https://user-images.githubusercontent.com/1509205/78934315-5806fc00-7a70-11ea-9085-4d04d1b73df7.png">

## Solution
* To accomplish this the Material Design Bottom Sheet component was utilized. It's already used within the app, so I simply created a BottomSheet dialog that would house the options/actions the user can perform before posting. 

* The nudge is supposed to be triggered when the user is publishing a post for the first time so to make testing possible I modified the `publishNow` call sites within `EditPostActivity`. b5b663bac97fcafed08fd5cb4487af298187d62b

* To make the testing experience more functionality, I decided to create the first fragment that will contain the actions the user performs such as adding tags, changing visibility/scheduling. Currently, this fragment is hardcoded into the `Bottomsheet` layout. In another PR #11634, I will replace this with a `FrameLayout` that will be the container for all fragments.

* The prepublishing actions UI is only being rendered at this point in a barebones way and requires more styling. There are no click listeners as yet as that will also be covered in the navigation PR #11634. 

* The prepublishing actions UI has a state-based approach and the rest of the UI for this screen such as the header, bottom divider and the publish/schedule button will be `UiStates`. These will also be covered later on. 

* The corners of the bottom sheet aren't being rendered with a round border-radius. This is also visible when the create FAB is clicked on the home screen and the bottom sheet is selected. There is a note on the board to [track this](https://github.com/orgs/wordpress-mobile/projects/65#card-36038752). 

* `PrepublishingViewModel.kt` does nothing right now but I had created it as part of the project setup. 

N.B Even though the PR has a lot of files, most of them are just boilerplate view models, adapters, etc. because this is the start of this feature/project 😉 

## Testing

1. Create a new post.
2. Click publish.
3. You should then see the bottom sheet. 

Dark Mode | Light Mode 
--------|-------
 <img src="https://user-images.githubusercontent.com/1509205/78941829-082f3180-7a7e-11ea-81c7-542e079671c2.png" width="320">  |  <img src="https://user-images.githubusercontent.com/1509205/78941832-09605e80-7a7e-11ea-9cee-9b86319731de.png" width="320">


## Reviewing 
Only 1 reviewer is needed but anyone can review. 

## Submitter Checklist

- [x] If it's feasible, I have added unit tests. user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.
